app/webpacker/components/CompetitionForm/FormSections/RegistrationFees.js

```diff
@@ -28,8 +30,11 @@ export default function RegistrationFees() {
     entryFees,
     competitorLimit,
     registration,
+    paymentInformation,
   } = useFormObject();
 
+  const [useWCAPayment, setUseWCAPayment] = useCheckboxState(Boolean(paymentInformation));
```

Comment:
Hm, I think it is quite dangerous to implicitly derive a boolean like that. What if the user starts to fill out, checks the box, then realizes it makes another new field pop up, starts to chat to the organizer about this, they pospone filling it out, save it temporarily (which isn't even guaranteed to work, see other comment above) and then reload the page only to find out the check mark hasn't been persisted?

This may be perceived as a "bug" and reported to us many times, because people are used to everything in the form being persisted properly.